### DISCUSSION TITLE
unable to create migration PlanetScale

### DISCUSSION THREAD

**Discussion by saihaj** [prisma/prisma#9703] [Q&A] [answered]
I am using planetscale and I am trying to create new migrations. I created 2 branches one feature and other shadow. I `yarn prisma migrate status` and apply the last one and then I do `yarn prisma migrate dev` and I get the following:
```zsh
yarn run v1.22.10
$ /Users/saihajpreetsingh/Desktop/scribedao/retweet/node_modules/.bin/prisma migrate dev
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "scribedao-prod" at "hbr7v1ewleao.us-east-3.psdb.cloud:3306"

Error: P3006

Migration `20210907234126_` failed to apply cleanly to the shadow database. 
Error code: P3018
Error:
A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 20210907234126_

Database error code: 1317

Database error:
Foreign keys cannot be created on this database. Learn more how to handle this: https://pris.ly/d/migrate-no-foreign-keys

Please check the query number 8 from the migration file.

```

Prisma version:
```
yarn run v1.22.10
$ /Users/saihajpreetsingh/Desktop/scribedao/retweet/node_modules/.bin/prisma --version
Environment variables loaded from .env
prisma                  : 3.0.1
@prisma/client          : 3.0.1
Current platform        : darwin
Query Engine (Node-API) : libquery-engine 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at node_modules/prisma/node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at node_modules/prisma/node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at node_modules/prisma/node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at node_modules/prisma/node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash    : 2452cc6313d52b8b9a96999ac0e974d0aedf88db
Studio                  : 0.423.0
Preview Features        : planetScaleMode
```

**Comment by ryands17** [comment, marked as answer]
@saihaj 👋 

Currently we don't recommend using `prisma migrate` on Planetscale. Also it seems that you are using foreign keys and Planetscale doesn't support that. The way to tell migrate not to create foreign keys is adding the `referentialIntegrity` preview feature and using `referentialIntegrity = "prisma"`  in the datasource that you can read about [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-integrity).


The best way to go for now is using `db push` when using PlanetScale and not using foreign keys.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
